cirq/pasqal/pasqal_sampler.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+
+import time
+import requests
+
+import cirq
+
+
+class PasqalSampler(cirq.work.Sampler):
+
+    def __init__(self, remote_host: str, access_token: str = '') -> None:
+        """
+        Args:
+            remote_host: Address of the remote device.
+            access_token: Access token for the remote api.
+        """
+        self.remote_host = remote_host
+        self._authorization_header = {"Authorization": access_token}
+
+    def _serialize_circuit(self, circuit: cirq.circuits.Circuit,
+                           param_resolver: cirq.study.ParamResolverOrSimilarType
+                          ) -> str:
+        """Serialize a given Circuit.
+        Args:
+            circuit: The circuit to be run
+            param_resolver: Param resolver for the
+        Returns:
+            json serialized string
+        """
+        circuit = cirq.protocols.resolve_parameters(circuit, param_resolver)
+        serialized_circuit = cirq.to_json(circuit)
+
+        return serialized_circuit
+
+    def _retrieve_serialized_result(self, task_id: str) -> str:
+        """Retrieves the results from the remote Pasqal device
+        Args:
+            task_id: id of the current task.
+        Returns:
+            json representation of the results
+        """
+
+        url = f'{self.remote_host}/get-result/{task_id}'
+        while True:
+            response = requests.get(
+                url,
+                headers=self._authorization_header,
+                verify=False,
+            )
+            response.raise_for_status()
+
+            result = response.text
+            if result:
+                return result
+
+            time.sleep(1.0)  # coverage: ignore
```

Comment:
Why are we sleeping here? Why the coverage ignore?

Reply:
So that the get request to retrieve the result is executed every second. Would you have another suggestion ?

Reply:
Okay makes sense. Yeah, coverage didn't need to be ignored here since this line will execute every time `_retrieve_serialized_result` is called.